charts/shoot-cloud-config/templates/scripts/_cloud-config-script.sh

```diff
@@ -104,4 +105,27 @@ if ! diff "$PATH_CLOUDCONFIG" "$PATH_CLOUDCONFIG_OLD" >/dev/null; then
 fi
 
 rm "$PATH_CLOUDCONFIG"
+
+ANNOTATION_RESTART_SYSTEMD_SERVICES="worker.gardener.cloud/restart-systemd-services"
+
+# Try to find Node object for this machine
+if [[ -f "$DIR_KUBELET/kubeconfig-real" ]]; then
+  {{`NODE="$(/opt/bin/kubectl --kubeconfig="$DIR_KUBELET/kubeconfig-real" get node -o go-template="{{ range \$i, \$item := .items }}{{ range \$j, \$address := \$item.status.addresses }}{{ if and (eq \$address.type \"Hostname\") (eq \$address.address \"$(hostname)\") }}{{ \$item.metadata.name }}{{ if (index \$item.metadata.annotations \"$ANNOTATION_RESTART_SYSTEMD_SERVICES\") }} {{ index \$item.metadata.annotations \"$ANNOTATION_RESTART_SYSTEMD_SERVICES\" }}{{ end }}{{ end }}{{ end }}{{ end }}")"`}}
+
+  if [[ ! -z "$NODE" ]]; then
+    NODENAME="$(echo "$NODE" | awk '{print $1}')"
+    SYSTEMD_SERVICES_TO_RESTART="$(echo "$NODE" | awk '{print $2}')"
+  fi
+
+  # Update checksum/cloud-config-data annotation on Node object if possible
+  if [[ ! -z "$NODENAME" ]] && [[ -f "$PATH_CHECKSUM" ]]; then
+    /opt/bin/kubectl --kubeconfig="$DIR_KUBELET/kubeconfig-real" annotate node "$NODENAME" "checksum/cloud-config-data=$(cat "$PATH_CHECKSUM")" --overwrite
+  fi
+
+  # Restart systemd services if requested
+  for service in $(echo "$SYSTEMD_SERVICES_TO_RESTART" | sed "s/,/ /g"); do
+    echo "Restarting systemd service $service due to $ANNOTATION_RESTART_SYSTEMD_SERVICES annotation"
+    systemctl restart "$service"
```

Comment:
To make it safer to use, we could check first if the systemd service actually exists. 
Otherwise, the script will silently fail and not remove the annotation. 
So every 30 seconds the services will be restarted. 
In the worst-case someone specified something like "kubelet, dockers" which will kill the node due to a spelling mistake.

Reply:
Yeah, nice catch. What about something like 
```suggestion
    systemctl restart "$service" || true
```
to prevent that such mistake can lead to the annotation not being removed?

Reply:
sure, good idea